manifests/plugin/discovery.pp

```diff
@@ -4,44 +4,51 @@
 #
 # === Parameters:
 #
-# $version::         version string of discovery image, in form of x.y.z-r
+# $major_version::   version string of discovery image, in form of x.y
 #
-# $source::          mirror url from which the image files should be obtained, you
-#                    can use http(s):// or file://
-#
-# $initrd::          name of initrd image file
-#
-# $kernel::          name of kernel file
+# $minor_version::   version string of discovery image, in form of x.y.z
 #
 # $install_images::  should the installer download and setup discovery images
 #                    for you? the average size is few hundreds of MB
 #                    type:boolean
 #
+# $tftp_root::       tftp root to install image into (optional)
+#
+# $source_url::      source URL to download from (optional)
+#
 class foreman::plugin::discovery (
-  $version        = $foreman::plugin::discovery::params::version,
-  $source         = $foreman::plugin::discovery::params::source,
-  $initrd         = $foreman::plugin::discovery::params::initrd,
-  $kernel         = $foreman::plugin::discovery::params::kernel,
+  $major_version  = $foreman::plugin::discovery::params::major_version,
+  $minor_version  = $foreman::plugin::discovery::params::minor_version,
   $install_images = $foreman::plugin::discovery::params::install_images,
+  $tftp_root      = $foreman::plugin::discovery::params::tftp_root,
+  $source_url     = "http://downloads.theforeman.org/discovery/releases/latest/",
 ) inherits foreman::plugin::discovery::params {
 
+  $tftp_root_clean = regsubst($tftp_root, '/$', '')
   validate_bool($install_images)
+  validate_absolute_path($tftp_root_clean)
+
+  $image_name      = "fdi-image-${minor_version}.tar"
 
   foreman::plugin {'discovery':
   }
 
   if $install_images {
-    include ::tftp::params
 
-    foreman::remote_file {"${::tftp::params::root}boot/${kernel}":
-      remote_location => "${source}${kernel}",
+    foreman::remote_file {"${tftp_root_clean}/boot/${image_name}":
+      remote_location => "${source_url}${image_name}",
       mode            => 0644,
-      require         => File["${::tftp::params::root}boot"],
+      require         => File["${tftp_root_clean}/boot"],
```

Comment:
I wonder we should solve this straight away in foreman::remote_file, but can be fixed later without breaking the API.